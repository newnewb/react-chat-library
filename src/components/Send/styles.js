// @ts-check

// Ours
import Colors from '../../colors';

const styles = {
	container: {
		height: 44,
		justifyContent: 'flex-end'
	},
	text: {
		color: Colors.defaultBlue,
		fontWeight: '600',
		fontSize: 17,
		backgroundColor: Colors.backgroundTransparent,
		marginBottom: 12,
		marginLeft: 10,
		marginRight: 10
	}
};

export default styles;
