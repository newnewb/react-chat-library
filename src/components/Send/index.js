// @ts-check

// Packages
import React from 'react';

// Ours
import { CSS } from '../../utils';
import Context from '../../context';
import styles from './styles';

const Send = props => (
	<Context.Consumer>
		{ctx => {
			// Only display the button if `alwaysShow` is set or the input text is
			// Not empty.
			let visibility = {};
			if (!props.alwaysShow && ctx.input.text.trim().length <= 0) {
				visibility = { display: 'none' };
			}

			return (
				<button
					type="button"
					data-test-id="send"
					style={CSS([styles.container, props.containerStyle, visibility])}
					onClick={() => {
						props.onSend(ctx.input.text.trim());
					}}
				>
					<div>
						{props.children || (
							<span style={CSS([styles.text, props.textStyle])}>
								{props.label}
							</span>
						)}
					</div>
				</button>
			);
		}}
	</Context.Consumer>
);

Send.defaultProps = {
	onSend: () => {},
	label: 'Send',
	containerStyle: {},
	textStyle: {},
	children: null,
	alwaysShow: false
};

export default Send;
