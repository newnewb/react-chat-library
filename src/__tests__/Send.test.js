import React from 'react';
import renderer from 'react-test-renderer';

import Send from '../components/Send';

it('should render <Send /> and compare with snapshot', () => {
	const component = renderer.create(<Send />);
	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});
