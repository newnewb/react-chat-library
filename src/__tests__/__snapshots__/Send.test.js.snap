// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <Send /> and compare with snapshot 1`] = `
<button
  data-test-id="send"
  onClick={[Function]}
  style={
    Object {
      "display": "none",
      "height": 44,
      "justifyContent": "flex-end",
    }
  }
  type="button"
>
  <div>
    <span
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#0084ff",
          "fontSize": 17,
          "fontWeight": "600",
          "marginBottom": 12,
          "marginLeft": 10,
          "marginRight": 10,
        }
      }
    >
      Send
    </span>
  </div>
</button>
`;
